### PR TITLE
[Horizontal Bar Chart With Axis] Enable multiple legend selection

### PR DESCRIPTION
**Pull request by srmukher**
[Horizontal Bar Chart With Axis] Enable multiple legend selection

### CODE REVIEW

File: packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx [hunk 1]
```diff
@@ -54,6 +54,7 @@ export interface IHorizontalBarChartWithAxisState extends IBasestate {
   callOutAccessibilityData?: IAccessibilityProps;
```
Comment (fabricteam) [inline packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx:54]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv8 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33484&commitId=eeeb01d30d871ae7da8a447de85de94f9d9ba9ce&env=prod&clientType=FLUENTUI)


  <details><summary><b>Callout</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Callout.Bottom auto edge.chromium.png|2309| Changed|
|Callout.Left top edge.chromium.png|1949| Changed|
|Callout.No callout width specified.chromium.png|2319| Changed|

</details>


  <details><summary><b>Keytip</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Keytip.Offset.chromium.png|121| Changed|

</details>


  <details><summary><b>react-charting-LineChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-LineChart.Gaps.chromium.png|1| Changed|

</details>


  <details><summary><b>react-charting-VerticalBarChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-VerticalBarChart.Basic - Secondary Y Axis.chromium.png|4| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx [hunk 2]
```diff
@@ -780,11 +760,61 @@ export class HorizontalBarChartWithAxisBase extends React.Component<
...
+  private _onLegendChange = (
+    selectedLegends: string[],
+    event: React.MouseEvent<HTMLButtonElement>,
+    currentLegend?: ILegend,
+  ) => {
+    this.setState({ selectedLegends });
+    if (this._isLegendSelected()) {
+      if (this._isSpecificLegendTitleSelected(currentLegend?.title!)) {
+        this.setState({
+          isLegendSelected: false,
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx:799]: >isLegendSelected: false [](http://example.com/codeflow?start=10&length=23)

why setting this to false if setting selectedLegendTitle to a value

File: packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx [hunk 3]
```diff
@@ -780,11 +760,61 @@ export class HorizontalBarChartWithAxisBase extends React.Component<
         focusZonePropsInHoverCard={this.props.focusZonePropsForLegendsInHoverCard}
         overflowText={this.props.legendsOverflowText}
         {...this.props.legendProps}
+        canSelectMultipleLegends={this.props.legendProps?.canSelectMultipleLegends}
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx:763]: > canSelectMultipleLegends={this.props.legendProps?.canSelectMultipleLegends} [](http://example.com/codeflow?start=7&length=76)

this line is not needed

File: packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx [hunk 4]
```diff
@@ -378,10 +383,7 @@ export class HorizontalBarChartWithAxisBase extends React.Component<
     refArrayIndexNumber: number,
     color: string,
   ): void => {
-    if (
-      this.state.isLegendSelected === false ||
-      (this.state.isLegendSelected && this.state.selectedLegendTitle === point.legend)
-    ) {
+    if (!this._isLegendSelected() || (this._isLegendSelected() && this._isSpecificLegendTitleSelected(point.legend!))) {
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx:386]: >(!this._isLegendSelected() || (this._isLegendSelected() && this._isSpecificLegendTitleSelected(point.legend!))) { [](http://example.com/codeflow?start=7&length=113)

shouldn't bar hover and bar focus conditions be same?

File: packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx [hunk 5]
```diff
@@ -456,8 +458,9 @@ export class HorizontalBarChartWithAxisBase extends React.Component<
 
     const bars = sortedBars.map((point: IHorizontalBarChartWithAxisDataPoint, index: number) => {
       let shouldHighlight = true;
-      if (this.state.isLegendHovered || this.state.isLegendSelected) {
-        shouldHighlight = this.state.selectedLegendTitle === point.legend;
+      if (this._isLegendHovered() || this._isLegendSelected()) {
+        shouldHighlight =
+          this._isSpecificLegendTitleSelected(point.legend!) || this._isSpecificLegendSelected(point.legend!);
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx:463]: >this._isSpecificLegendTitleSelected(point.legend!) || this._isSpecificLegendSelected(point.legend!); [](http://example.com/codeflow?start=10&length=100)

same here?

File: packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx [hunk 6]
```diff
@@ -584,8 +587,8 @@ export class HorizontalBarChartWithAxisBase extends React.Component<
     const { useSingleColor = false } = this.props;
     const bars = this._points.map((point: IHorizontalBarChartWithAxisDataPoint, index: number) => {
       let shouldHighlight = true;
-      if (this.state.isLegendHovered || this.state.isLegendSelected) {
-        shouldHighlight = this.state.selectedLegendTitle === point.legend;
+      if (this._isLegendHovered() || this._isLegendSelected()) {
+        shouldHighlight = this._isSpecificLegendTitleSelected(point.legend!);
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx:591]: and here?

**Comment by AtishayMsft** [comment]
Change file is missing

File: packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx (hunk 3 above)
Comment (srmukher) [reply packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx:763]: removed

File: packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx (hunk 4 above)
Comment (srmukher) [reply packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx:386]: updated

File: packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx (hunk 5 above)
Comment (srmukher) [reply packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx:463]: done

File: packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx (hunk 6 above)
Comment (srmukher) [reply packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx:591]: updated

File: packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx (hunk 2 above)
Comment (srmukher) [reply packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx:799]: updated

**Comment by AtishayMsft** [comment]
CI build is failing

File: packages/react/etc/react.api.md [hunk 7]
```diff
@@ -1604,49 +1604,49 @@ export { FabricPerformance }
 // @public (undocumented)
 export enum FabricSlots {
     // (undocumented)
-    black = 21,// BaseSlots.primaryColor, Shade[Shade.Unshaded]);
+    black = 21,
```
Comment (AtishayMsft) [inline packages/react/etc/react.api.md:1607]: revert this change

Comment (srmukher) [reply packages/react/etc/react.api.md]: done

**Review by AtishayMsft — APPROVED** [review]

**Comment by AtishayMsft** [comment]
Component did update change is pending

**Review by AtishayMsft — CHANGES_REQUESTED** [review]
Component did update change in pending

**Comment by srmukher** [comment]
> Component did update change in pending

Added the change

**Review by AtishayMsft — APPROVED** [review]
